src/aap_eda/api/views/credential.py

```diff
@@ -117,3 +126,25 @@ def get_serializer_class(self):
 
     def get_response_serializer_class(self):
         return serializers.CredentialSerializer
+
+    def destroy(self, request, *args, **kwargs):
+        force = request.query_params.get("force", "false").lower() in [
+            "true",
+            "1",
+            "yes",
+        ]
+        credential = self.get_object()
+
+        # If the credential is in use and the 'force' flag
+        # is not True, raise a PermissionDenied exception
+        is_used = models.Activation.objects.filter(
+            decision_environment__credential=credential
+        ).exists()
+
+        if is_used and not force:
+            raise exceptions.Forbidden(
+                "Credential is being used by Activations "
+                "and cannot be deleted. If you want to force delete, "
+                "please add /?force=true query param."
+            )
+        return super().destroy(request, *args, **kwargs)
```

Comment:
Replacing it with the following would save us calling `self.get_object()` once again
```
self.perform_destroy(instance)
return Response(status=status.HTTP_204_NO_CONTENT)
```
I think `self.get_object()` makes another query to fetch the object (which we already have, `credential`), which is redundant. Thoughts?

Reply:
The point to me was not breaking down the process flow and continue with the flow of the superclass. Override a specific operation doesn't need necessarily to replace the whole common logic. Efficiency vs extensibility and maintenance.

Reply:
changed